core/server/src/main/java/alluxio/master/file/meta/InodeFile.java

```diff
@@ -296,6 +287,31 @@ public static InodeFile fromJournalEntry(InodeFileEntry entry) {
     return inode;
   }
 
+  /**
+   * Creates an {@link InodeFile}.
+   *
+   * @param id id of this inode
+   * @param parentId id of the parent of this inode
+   * @param name name of this inode
+   * @param fileOptions options to create this file
+   * @return the {@link InodeFile} representation
+   */
+  public static InodeFile create(long id, long parentId, String name,
```

Comment:
should we change the constructor to private to enforce that all objects are created using this factory?